the-collab-lab/tcl-12-smart-shopping-list
# Add item to list and select frequency of purchase

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- Users can add items to a list and select purchase frequency 
- When a user adds an item to the list, the Firestore database updates the list associated with the users token
- Display items from the list in the list view on our application
- Installed `react-hooks-form` dependency 
- Added validations and error handling


## Related Issue

Issue #4 

## Acceptance Criteria

-User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
-When the user submits the form, the item is saved to the database, associated with the user’s token
-Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
-The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
-Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓   | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![tcl-before](https://user-images.githubusercontent.com/61370161/96019858-c1e2d280-0e12-11eb-97c9-49b3368ac008.png)


### After

![Screen Shot 2020-10-14 at 11 46 10 AM](https://user-images.githubusercontent.com/61370161/96020041-15edb700-0e13-11eb-8d47-5bb36f620872.png)

![Screen Shot 2020-10-14 at 11 46 25 AM](https://user-images.githubusercontent.com/61370161/96020071-1c7c2e80-0e13-11eb-8ce0-d7a23e77098e.png)

![Screen Shot 2020-10-14 at 11 47 14 AM](https://user-images.githubusercontent.com/61370161/96020082-21d97900-0e13-11eb-8014-1144c66c1459.png)


## Testing Steps / QA Criteria

- Go to the branch (`git checkout tc-eb-add-item-to-list`) and pull it down (`git pull origin tc-eb-add-item-to-list`)
- Run `npm install`
- Open Firestore database and go to the lists collection and the test-token document
- Go to "Add an Item". Add item name and select frequency then submit. Database and list view should automatically update. You should receive an alert when item is submitted.
- Make a change in the Firestore database to view real time updates on the application
- Try to submit a blank item to view validation error
